content/en/blog/_posts/2025-02-14-cloud-controller-manager-chicken-egg-problem.md

```diff
@@ -0,0 +1,124 @@
+
+As we can see in the following diagram, when the **kubelet** starts, it registers the `Node` object with the apiserver, Tainting the node so it can be processed first by the cloud-controller-manager. The initial `Node` is missing the cloud-provider specific information, like the Node Addresses and the Labels with the cloud provider specific information like the Node, Region and Instance type information.
+
+![Diagram of Node initialization](https://mermaid.ink/img/pako:eNqVU01PAjEQ_StNTyrLChgObgyJwYtRiPEjXvZStsPSuNvitMUg4b87u-VjIRLjHjad9nXemzfTFc-MBJ5wC58edAZ3SuQoylQz-oR3RvtyAhhihMwxzCdn3etuxHq9K_r1--fh8MFPoADXHgxa1bIt5soCLgATNkQQDtiYqAJ2bCg0dLa9lbBXobRLbiZ4OWCagHFWGC_naBZKAsYfhEMNDmyszJ5wz0K87V2yioltaGVAg5a_cTdl1reUrRDsGYRcBjW1MpARGylrlc4D7lZKBGvBXkQsjuO_cr8QP3ubSxJkj8zsdToR6_Y7x2Ye1NYaVna0M6MdmqKgvVJokVep34XLZqQevoKyg7L3ik4nuNfKKVGo79Ci0IQazp42_rNHQdba6Kj2Rumn0m_70jQj-NCYh380p25MqnnES8BSKEnDu6oup9zNoISUJ7SUMBW-cClP9Zqg1SC_LHXGE4ceIo7G5zOeTEVhKfK1nM3k73ZBKmdwFJ5H_UrWPzszEMk?type=png\)]\(https://mermaid.live/edit#pako:eNqVU01PAjEQ_StNTyrLChgObgyJwYtRiPEjXvZStsPSuNvitMUg4b87u-VjIRLjHjad9nXemzfTFc-MBJ5wC58edAZ3SuQoylQz-oR3RvtyAhhihMwxzCdn3etuxHq9K_r1--fh8MFPoADXHgxa1bIt5soCLgATNkQQDtiYqAJ2bCg0dLa9lbBXobRLbiZ4OWCagHFWGC_naBZKAsYfhEMNDmyszJ5wz0K87V2yioltaGVAg5a_cTdl1reUrRDsGYRcBjW1MpARGylrlc4D7lZKBGvBXkQsjuO_cr8QP3ubSxJkj8zsdToR6_Y7x2Ye1NYaVna0M6MdmqKgvVJokVep34XLZqQevoKyg7L3ik4nuNfKKVGo79Ci0IQazp42_rNHQdba6Kj2Rumn0m_70jQj-NCYh380p25MqnnES8BSKEnDu6oup9zNoISUJ7SUMBW-cClP9Zqg1SC_LHXGE4ceIo7G5zOeTEVhKfK1nM3k73ZBKmdwFJ5H_UrWPzszEMk)
+
+This new initialization process adds some latency to the node readiness. Previously, the kubelet was able to initialize the node at the same time it created the node. Since the logic has moved to the cloud-controller-manager, this can cause a [chicken and egg problem](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/#chicken-and-egg) during the cluster bootstrapping for those Kubernetes architectures that do not deploy the controller manager as the other components of the control plane, commonly as static pods, standalone binaries or daemonsets/deployments with tolerations to the taints and using `hostNetwork` (more on this below)
+
+## Examples of the dependency problem
+
+As noted above, it is possible during bootstrapping for the cloud-controller-manager to be unschedulable and as such the cluster will not initialize properly. The following are a few concrete examples of how this problem can be expressed and the root causes for why they might occur.
+
+These examples assume you are running your cloud-controller-manager using a Kubernetes resource (e.g. `Deployment`, `DaemonSet`, or similar) to control its lifecycle. Because these methods rely on Kubernetes to schedule the cloud-controller-manager, care must be taken to ensure it will schedule properly.
```

Comment:
Per the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/): no backticks here.